### PR TITLE
Docs: Update the Triage documentation for accuracy and general content edits.

### PR DESCRIPTION
**Pull request by ndiego**
## What? & Why?
The PR thoroughly audits the [Triage](https://developer.wordpress.org/block-editor/contributors/triage/) documentation in the Block Editor Handbook to ensure it is up-to-date and approachable for new and long-time contributors alike.

### CODE REVIEW

**Review by alexstine — APPROVED** [review]
👍

File: docs/contributors/triage.md [hunk 1]
```diff
@@ -1,24 +1,23 @@
-### Get involved in triage
+The Gutenberg repository needs to be triaged regularly to keep it healthy. Triage is the practice of reviewing existing [issues](https://github.com/WordPress/gutenberg/issues) and [pull requests](https://github.com/WordPress/gutenberg/pulls) to ensure they're relevant, actionable, and have all the information they need. Anyone can help triage, although you will need to be a member of the Triage Team to edit the title of an issue or modify its labels.
 
-To keep the repository healthy, it needs to be triaged regularly. Triage is the practice of reviewing existing issues and pull requests to make sure they’re relevant, actionable, and have all the information they need. Anyone can help triage, although you’ll need to be a member of the triage team for the Gutenberg repository to modify an issue’s labels or edit its title.
+## Joining the triage team
 
-### Join the triage team
+The [Triage Team](https://github.com/orgs/WordPress/teams/gutenberg-triage-team/members) is an open group of contributors that ensure triage is done consistently across the Gutenberg repository. Various types of triage happen:
```
Comment (hanneslsm) [inline docs/contributors/triage.md:5]: that the link does not work is because this PR is a draft, right?

Comment (ndiego) [reply docs/contributors/triage.md:5]: Good catch, it looks like you already need to be part of the team to access. I will get that fixed.

**Comment by ndiego** [comment]
This PR is outdated, so I am going to close. I'll open a new one if needed.

**Pull request closed without merging**